stacker/providers/aws/default.py

```diff
@@ -1,20 +1,185 @@
 import json
 import logging
 import os
-import time
-
-import botocore.exceptions
-
+from stacker.exceptions import NotInitialized
+import re
+import botocore
+import uuid
 from ..base import BaseProvider
 from ... import exceptions
 from ...util import retry_with_backoff
 from stacker.session_cache import get_session
+from stacker.status import (
+    SubmittedStatus,
+    CompleteStatus,
+)
 
-logger = logging.getLogger(__name__)
 
+logger = logging.getLogger(__name__)
 MAX_TAIL_RETRIES = 5
 
 
+class Message(object):
+
+    def __init__(self, metadata, parent_queue):
+
+        self._parent_queue = parent_queue
+
+        parsed_message = self.parse_message(metadata)
+
+        for key, value in parsed_message.iteritems():
+            setattr(self, key, value)
+
+        for key, value in metadata.iteritems():
+            setattr(self, key, value)
+
+    def parse_message(self, message):
```

Comment:
Not sure what the idea is behind this - should it be returning a Message object?  If so, should this be a `classmethod`?